src/ucp/api/ucp.h

```diff
@@ -838,22 +838,40 @@ ucs_status_t ucp_worker_wait(ucp_worker_h worker);
  * by calling @ref ucp_worker_signal .
  *
  * @code {.c}
- * status = ucp_worker_get_efd (worker, &fd);
- * while (1) {
- *     ucp_worker_arm (worker);
- *     poll (&fd, nfds, timeout);
- *     <handle event>
+ * void application_initialization() {
+ *     ...
+ *     status = ucp_worker_get_efd(worker, &fd);
+ *     ...
+ * }
+ * void process_comminucation() {
+ *     for (;;) {
+ *         ucp_worker_progress(worker);
+ *         check_for_events();              // receive() operations
+ *         status = ucp_worker_arm(worker); // arm the worker and clean-up fd
+ *         if (UCS_OK == status) {
+ *             poll(&fds, nfds, timeout);   // wait for events
+ *         } else if (UCS_ERR_BUSY == status) {
+ *             continue;                    // poll for more events
+ *         } else {
+ *             abort();
+ *         }
+ *     }
  * }
- *
- * - The code in this example, could be replaced by ucp_worker_wait.
  * @endcode
  *
  * @note UCP @ref ucp_feature "features" have to be triggered
  *   with @ref UCP_FEATURE_WAKEUP to select proper transport
  *
  * @param [in]  worker    Worker of notified events.
  *
- * @return Error code as defined by @ref ucs_status_t
+ * @return ::UCS_OK        The operation completed successfully. File descriptor
+ *                         will be signaled by new events.
+ * @return ::UCS_ERR_BUSY  There are unprocessed events which prevent the
+ *                         file descriptor from being armed. These events should
+ *                         be removed by calling @ref ucp_worker_progress().
+ *                         The operation is not completed. File descriptor
+ *                         will not be signaled by new events.
+ * @return @ref ucs_status_t "Other" different error codes in case of issues.
```

Comment:
@shamisp The reference to the example code in ucp_hello_world.c has implemented already. The `ucp.h` has doxygen `example` clause and it provides paragraph `examples` automatically for all referenced functions. It appears in html generated documentation. This is one more reason to have generated html documentation online to see all doxygen magic.